frontend/packages/console-shared/src/hooks/scroll.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+export enum ScrollDirection {
+  scrollingUp = 'scrolling-up',
+  scrollingDown = 'scrolling-down',
+  scrolledToBottom = 'scrolled-to-bottom',
+  scrolledToTop = 'scrolled-to-top',
+}
+
+export const useScrollDirection = (node: HTMLElement) => {
+  const scrollPosition = React.useRef(null);
+  const [scrollDirection, setScrollDirection] = React.useState<ScrollDirection>(null);
+  const handleScroll = React.useCallback(
+    (event) => {
+      const { scrollHeight, scrollTop, clientHeight } = event.target;
+      if (scrollPosition.current) {
+        if (scrollHeight - scrollTop === clientHeight) {
+          setScrollDirection(ScrollDirection.scrolledToBottom);
+        } else if (scrollTop === 0) {
+          setScrollDirection(ScrollDirection.scrolledToTop);
+        } else if (
+          scrollPosition.current > scrollTop &&
+          scrollDirection !== ScrollDirection.scrollingUp
+        ) {
+          setScrollDirection(ScrollDirection.scrollingUp);
+        } else if (
+          scrollPosition.current < scrollTop &&
+          scrollDirection !== ScrollDirection.scrollingDown
+        ) {
+          setScrollDirection(ScrollDirection.scrollingDown);
+        }
+      }
+      scrollPosition.current = scrollTop;
+    },
+    [scrollDirection],
+  );
+
+  React.useEffect(() => {
+    if (node) {
+      node.addEventListener('scroll', (event) => setTimeout(() => handleScroll(event), 100));
+    }
+    return () => {
+      node &&
+        node.removeEventListener('scroll', (event) => setTimeout(() => handleScroll(event), 100));
```

Comment:
This will not work. Two different references. Perhaps local scope the function to a variable and use that?

Reply:
Done